src/lib/widgets/layers-toggle/sub-layers-toggle.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit, Input} from '@angular/core';
+
+@Component({
+  selector: 'esri-sublayersToggle',
+  template: `<!-- Only display if there is a layer with sublayers -->
+    <div *ngIf="layer && layer.sublayers && layer.sublayers.items">
+        <div *ngFor="let sublayer of layer.sublayers.items.slice().reverse()" class="sublayers">
+            <input type="checkbox" [checked]="sublayer.visible" (click)="onCheck($event, sublayer)" /> {{sublayer.title}}
+            <esri-sublayersToggle [layer]="sublayer"></esri-sublayersToggle><!--recursive sub layers-->
+        </div>
+    </div>
+  `,
+  styles: [`
+    .sublayers {
+        padding-left: 15px;
+    }
+  `]
+})
+export class SubLayersToggleComponent {
+  @Input() layer: any;
```

Comment:
Please set a proper data type for all properties, variables, and function arguments.